src/server.c

```diff
@@ -2900,6 +2900,18 @@ client_context_t *homekit_server_accept_client(homekit_server_t *server) {
     const struct timeval rcvtimeout = { 10, 0 }; /* 10 second timeout */
     setsockopt(s, SOL_SOCKET, SO_RCVTIMEO, &rcvtimeout, sizeof(rcvtimeout));
 
+    const int yes = 1; /* enable sending keepalive probes for socket */
+    setsockopt(s, SOL_SOCKET, SO_KEEPALIVE, &yes, sizeof(int));
```

Comment:
Could you change `sizeof(int)` to `sizeof(yes)`. A little bit of defensive programming.